src/btree/bt_sync.c

```diff
@@ -147,12 +147,14 @@ __sync_ref_list_pop(WT_SESSION_IMPL *session, WT_REF_LIST *rlp, uint32_t flags)
 
     for (i = 0; i < rlp->entry; i++) {
         /*
-         * Mark the obsolete page dirty to let the reconciliation to remove the updates from page.
-         * The obsolete pages with overflow keys cannot be fast deleted, so marking them dirty will
-         * let them clean during reconciliation.
+         * Mark the obsolete page dirty to let the reconciliation to remove the updates from page,
+         * except if the page is already marked as empty by previous reconciliation. The obsolete
+         * pages with overflow keys cannot be fast deleted, so marking them dirty will let them
+         * clean during reconciliation.
          */
         WT_RET(__wt_page_modify_init(session, rlp->list[i]->page));
-        __wt_page_modify_set(session, rlp->list[i]->page);
+        if (rlp->list[i]->page->modify->rec_result != WT_PM_REC_EMPTY)
+            __wt_page_modify_set(session, rlp->list[i]->page);
```

Comment:
The phrase "let them clean during reconciliation" was confusing? Maybe reword for clarity?

Reply:
Fixed.